Docker compose startup regressions on node:25-alpine

## Summary
After pulling latest `main` and running `docker compose up -d --build`, runtime startup is unstable due to two Docker-related regressions:

1. `corepack` is not available in `node:25-alpine`, so Docker image build fails with the current `corepack enable && corepack prepare ...` steps.
2. The world client can enter a restart loop because host `packages/client/node_modules` can leak into the Docker build context, leaving stale pnpm symlink targets in the image.

## Reproduction
1. Pull latest `main`.
2. Run:
   ```bash
   docker compose -f docker-compose.yml up -d --build
   ```
3. Observe build/runtime behavior.

## Actual behavior
- Build step fails when `corepack` is invoked on `node:25-alpine`.
- Client container repeatedly restarts and cannot serve `:5173`.

Observed restart-loop log:
```text
Error: Cannot find module '/app/packages/client/node_modules/vite/bin/vite.js'
  code: 'MODULE_NOT_FOUND'
```

## Expected behavior
- `server` and `client` containers should build and start cleanly on latest `main`.
- `http://127.0.0.1:5173/` should be reachable without container restarts.

## Suspected root cause
- Dockerfiles assume `corepack` exists in `node:25-alpine`.
- `.dockerignore`/context rules allow workspace-local `node_modules` content to interfere with container-installed pnpm dependencies.

## Impact
Blocks local and CI-like Docker-based bring-up on latest code and makes world client unavailable.

--- a/.dockerignore
+++ b/.dockerignore
@@ -25,3 +25,10 @@ packages/plugin
 .github
 !packages/client
 !packages/client/**
+!packages/plugin
+!packages/plugin/**
+packages/client/node_modules
+packages/client/**/node_modules
+packages/plugin/node_modules
+packages/plugin/**/node_modules
+!tests/package.json

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,7 +1,7 @@
 # Build stage
 FROM node:25-alpine AS builder
 
-RUN corepack enable && corepack prepare pnpm@9.0.0 --activate
+RUN npm install -g pnpm@9.0.0
 
 WORKDIR /app
 
@@ -21,7 +21,7 @@ RUN pnpm --filter @openclawworld/shared build && \
 # Production stage
 FROM node:25-alpine AS production
 
-RUN corepack enable && corepack prepare pnpm@9.0.0 --activate
+RUN npm install -g pnpm@9.0.0
 
 RUN apk add --no-cache wget && \
     addgroup -g 1001 -S nodejs && \

--- a/Dockerfile.client
+++ b/Dockerfile.client
@@ -1,7 +1,7 @@
 FROM node:25-alpine
 
 RUN apk add --no-cache wget
-RUN corepack enable && corepack prepare pnpm@9.0.0 --activate
+RUN npm install -g pnpm@9.0.0
 
 # Create non-root user for security
 RUN addgroup -S appgroup && adduser -S appuser -G appgroup

